susemanager/bin/server-migrator.sh

```diff
@@ -19,27 +19,44 @@ echo
 read -n 1 -s -r -p "Press any key to start the migration or CTRL+C to cancel...";
 echo
 
+NEW_VERSION_ID=15.4
+
+CHECK_OS=$(cat /etc/os-release  | grep PRETTY_NAME)
+
+if [[ ${CHECK_OS} != *"openSUSE Leap"* ]]; then
+  echo "Please check your OS. openSUSE Leap is needed "
+  exit -1
+fi
+
 spacewalk-service stop
 rm -rf /etc/zypp/repos.d.old
 mv /etc/zypp/repos.d /etc/zypp/repos.d.old
 mkdir /etc/zypp/repos.d
-zypper ar -n "Main Repository" http://download.opensuse.org/distribution/leap/15.3/repo/oss repo-oss
-zypper ar -n "Main Update Repository" http://download.opensuse.org/update/leap/15.3/oss repo-update
-zypper ar -n "Non-OSS Repository" http://download.opensuse.org/distribution/leap/15.3/repo/non-oss repo-non-oss
-zypper ar -n "Update Repository (Non-Oss)" http://download.opensuse.org/update/leap/15.3/non-oss/ repo-update-non-oss
+zypper ar -n "Main Repository" http://download.opensuse.org/distribution/leap/${NEW_VERSION_ID}/repo/oss repo-oss
+zypper ar -n "Main Update Repository" http://download.opensuse.org/update/leap/${NEW_VERSION_ID}/oss repo-update
+zypper ar -n "Non-OSS Repository" http://download.opensuse.org/distribution/leap/${NEW_VERSION_ID}/repo/non-oss repo-non-oss
+zypper ar -n "Update Repository (Non-Oss)" http://download.opensuse.org/update/leap/${NEW_VERSION_ID}/non-oss/ repo-update-non-oss
 zypper ar -n "Uyuni Server Stable" https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable/images/repo/Uyuni-Server-POOL-x86_64-Media1/ uyuni-server-stable
-zypper ar -n "Update repository wiht updates from SUSE Linux Enterprise 15" http://download.opensuse.org/update/leap/15.3/sle repo-sle-update
-zypper ar -n "Update repository of openSUSE Backports" http://download.opensuse.org/update/leap/15.3/backports/ repo-backports-update
+zypper ar -n "Update repository wiht updates from SUSE Linux Enterprise" http://download.opensuse.org/update/leap/${NEW_VERSION_ID}/sle repo-sle-update
+zypper ar -n "Update repository of openSUSE Backports" http://download.opensuse.org/update/leap/${NEW_VERSION_ID}/backports/ repo-backports-update
 zypper ref
-zypper -n dup --allow-vendor-change
-if [ $? -ne 0 ];then
-    echo "Migration went wrong. Please fix the issues and try again."
+zypper -n dup
```

Comment:
I tried this script yesterday and I think we have conflicts which must be solved manual.
There is at least the "netty" problem, where we require an older version, but a patch require the newer version.

We also have a problem with the vendor of a product.

@juliogonzalez should we remove the `-n` and let the user make decissions?

Reply:
> I tried this script yesterday and I think we have conflicts which must be solved manual. 

What do you mean? Can you provide details? AFAIK @mbussolotto tested the upgrade and worked for him.

> There is at least the "netty" problem, where we require an older version, but a patch require the newer version.

Can you provide more details? What patch is requiring the newer versions? I remember you adjusted one spec to require *our* version, to fix a bug caused by a version bump at Leap 15.3

Wasn't that enough?

> We also have a problem with the vendor of a product.

You mean the discussion I have with Adrian? That's true, but doesn't really break the issue if we allow vendor change.

> @juliogonzalez should we remove the `-n` and let the user make decissions?

I don't see why? The script should just work, no questions asked.

And BTW: why are we removing `--allow-vendor-change`?

Reply:
> And BTW: why are we removing --allow-vendor-change?
I tested upgrade from JeOS image and it wasn't required, I will re-add it if you think it's safer

Reply:
But did the vendor change happened or not?

If it happened, then removing is OK.

Keep in mind there are a few packages we removed from our codestream and will now come from Leap, so those *must* change the vendor.